src/js/shared/config_file.js

```diff
@@ -153,26 +153,46 @@ const ConfigFile = {
     var searchProviders =
       (await LocalStore.getOne(StoreKey.SEARCH_PROVIDERS)) || [];
 
-    // - Check if the new list of search providers are already included on the list of menu items.
     var newProviders = _.map(newData.searchproviders, ConfigFile.parseProvider);
-    for (var i = 0; i < newProviders.length; i++) {
-      // Check if the search provider wasn't already included (and add it if not).
-      if (
-        !_.find(searchProviders, function (provider) {
-          return provider.link === newProviders[i].link;
-        })
-      ) {
-        searchProviders.push(newProviders[i]);
+
+    // - Check if the new list of search providers are already included on the list of menu items.
+    var mergeProviderOption = _.get(settings, "mergeSearchProviders", "0");
+    if (mergeProviderOption === "0") {
```

Comment:
Using numbers as values isn't very _dev-friendly_, `mergeProviderOption ` should have values "merge", "override" and "ignore" instead of "0", "1", "2"